quinn-proto/src/connection/streams/mod.rs

```diff
@@ -367,6 +385,28 @@ impl<'a> SendStream<'a> {
     }
 }
 
+struct WriteGuard<'a> {
```

Comment:
bikeshed: `WriteBuilder`? "Guard" usually means "cleans something up on Drop".

Reply:
Maybe `StreamWriter`? `Builder` invokes builder pattern for construction to me.